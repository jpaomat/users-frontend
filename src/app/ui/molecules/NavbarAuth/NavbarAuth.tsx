import { ButtonApp } from '../../atoms';
import PropTypes from 'prop-types';
import { INavbarAuth } from '../../interfaces';

export const NavbarAuth = ({ userName }: INavbarAuth) => {
	return (
		<ul className='navbar-nav ml-auto'>
			<span className='nav-item nav-link text-primary'>{userName}</span>
			<ButtonApp
				size='sm'
				variant='outline-light'
				onClick={() => console.log('Logout')}
			>
				<span>Logout</span>
			</ButtonApp>
		</ul>
	);
};

NavbarAuth.defaultProps = {
	userName: '',
};

NavbarAuth.propTypes = {
	userName: PropTypes.string.isRequired,
};
