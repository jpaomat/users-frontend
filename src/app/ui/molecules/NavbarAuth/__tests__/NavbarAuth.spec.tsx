import { fireEvent, render, screen } from '@testing-library/react';
import { NavbarAuth } from '../NavbarAuth';

describe('<NavbarAuth /> component', () => {

	beforeEach(()=> jest.clearAllMocks);
	
	test('should render component', () => {
		const { container } = render(
			<NavbarAuth userName='Maria' />
		);
		expect(container).toMatchSnapshot();
	});

	test('should execute onclick event', () => {
		render(
			<NavbarAuth userName='Maria' />
		);
		fireEvent.click(screen.getByText('Logout'));
		expect(screen.getByText('Logout')).toBeTruthy();
	});
});
