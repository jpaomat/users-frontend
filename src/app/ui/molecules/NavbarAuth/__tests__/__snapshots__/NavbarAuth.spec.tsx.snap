// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarAuth /> component should render component 1`] = `
<div>
  <ul
    class="navbar-nav ml-auto"
  >
    <span
      class="nav-item nav-link text-primary"
    >
      Maria
    </span>
    <button
      class="btn btn-outline-light btn-sm"
    >
      <span>
        Logout
      </span>
    </button>
  </ul>
</div>
`;
